app/com/gu/floodgate/reindex/ProgressTrackerController.scala

```diff
@@ -0,0 +1,43 @@
+package com.gu.floodgate.reindex
+
+import akka.actor.{ ActorRef, ActorLogging, Actor, Props }
+import com.gu.floodgate.contentsource.ContentSource
+import com.gu.floodgate.jobhistory.JobHistoryService
+import com.gu.floodgate.reindex.ProgressTracker.{ Cancel, TrackProgress }
+import com.gu.floodgate.reindex.ProgressTrackerController.{ RemoveTracker, LaunchTracker }
+import com.gu.floodgate.runningjob.{ RunningJob, RunningJobService }
+import com.typesafe.scalalogging.StrictLogging
+import play.api.libs.ws.WSAPI
+
+object ProgressTrackerController {
+  def props(ws: WSAPI, runningJobService: RunningJobService, jobHistoryService: JobHistoryService) = Props(new ProgressTrackerController(ws: WSAPI, runningJobService: RunningJobService, jobHistoryService))
+
+  case class LaunchTracker(contentSource: ContentSource, runningJob: RunningJob)
+  case class RemoveTracker(contentSource: ContentSource, runningJob: RunningJob)
+}
+
+class ProgressTrackerController(ws: WSAPI, runningJobService: RunningJobService, jobHistoryService: JobHistoryService) extends Actor with ActorLogging with StrictLogging {
+
+  private var runningTrackers: Map[String, ActorRef] = Map.empty[String, ActorRef]
```

Comment:
Yes, when we cancel a reindex via the API, we need to cancel the actor monitoring the progress.